src/spu/src/config/spu_config.rs

```diff
@@ -146,17 +146,17 @@ impl SpuConfig {
 }
 
 impl From<&SpuConfig> for ConfigOption {
-    fn from(config: &SpuConfig) -> ConfigOption {
+    fn from(config: &SpuConfig) -> Self {
         let log = &config.log;
-        ConfigOption::new(
-            log.base_dir.join(format!("spu-logs-{}", config.id)),
-            log.index_max_bytes,
-            log.index_max_interval_bytes,
-            log.segment_max_bytes,
-            log.flush_write_count,
-            log.flush_idle_msec,
-            log.max_batch_size,
-        )
+        ConfigOption::builder()
+            .base_dir(log.base_dir.join(format!("spu-logs-{}", config.id)))
+            .index_max_bytes(log.index_max_bytes)
+            .index_max_interval_bytes(log.index_max_interval_bytes)
+            .segment_max_bytes(log.segment_max_bytes)
+            .flush_write_count(log.flush_write_count)
+            .flush_idle_msec(log.flush_idle_msec)
+            .max_batch_size(log.max_batch_size)
+            .build()
```

Comment:
There is a pattern I have found with derive_builder that is really helpful when trying to collect options from other data structures or sources. We can implement functions directly on the `ConfigOptionBuilder` type, something like:

```rust
impl ConfigOptionBuilder {
    pub fn load_spu_config(&mut self, config: &SpuConfig) -> &mut Self {
        self
            .base_dir(log.base_dir.join(format!("spu-logs-{}", config.id)))
            .index_max_bytes(log.index_max_bytes)
            .index_max_interval_bytes(log.index_max_interval_bytes)
            .segment_max_bytes(log.segment_max_bytes)
            .flush_write_count(log.flush_write_count)
            .flush_idle_msec(log.flush_idle_msec)
            .max_batch_size(log.max_batch_size)
    }
}
```

This means that when constructing a `ConfigOption`, we can do something like start from a SpuConfig as a baseline, then override certain fields we want, all in the same builder flow. It'd look something like this:

```rust
let config = ConfigOption::builder()
    .load_spu_config(&spu_config)
    .base_dir("./something")
    // .other_stuff()
    .build();
```

If possible, I think I'd actually like to get rid of the `From<&SpuConfig> for ConfigOption` because I think implementing "From" a reference type is somewhat of an anti-pattern.